src/arrayMethodJoin.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    const sourceAarrayCopy = [...this];
+
+    for (let i = 0; i < sourceAarrayCopy.length; i++) {
+      if (sourceAarrayCopy[i] === null || sourceAarrayCopy[i] === undefined) {
+        sourceAarrayCopy[i] = '';
+      }
+    }
+
+    let divider = separator;
+
+    if (divider === undefined) {
+      divider = ',';
+    }
+
+    let string = '' + sourceAarrayCopy[0];
+
+    for (let i = 1; i < sourceAarrayCopy.length; i++) {
+      string += `${divider}${sourceAarrayCopy[i]}`;
+    }
```

Comment:
Dont need first line. You can move through loop and add separators to all element except last one.